### PR TITLE
nit(error_handling) - Added context to the UnexpectedIntegerOverflow error

### PR DESCRIPTION
**Pull request by wacban**

### CODE REVIEW

File: core/primitives/src/errors.rs [hunk 1]
```diff
@@ -53,7 +53,7 @@ impl From<InvalidTxError> for TxExecutionError {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum RuntimeError {
     /// An unexpected integer overflow occurred. The likely issue is an invalid state or the transition.
-    UnexpectedIntegerOverflow,
+    UnexpectedIntegerOverflow(String),
```
Comment (nagisa) [inline core/primitives/src/errors.rs:56]: There's exactly one location where `String` cannot be `&'static str` here and its in the `From` implementation just below. Interestingly that `From` implementation is precisely an implementation that fails to introduce *useful* context so I would remove it entirely and provide the necessary context where this implementation is used currently with `map_err`.

Comment (wacban) [reply core/primitives/src/errors.rs:56]: I was actually thinking in the exact opposite direction. The performance penalty for using strings is minimal but if we every want to add even more detail to the errors message we'd want the dynamic string. What do you think? I'm ok with either.

Comment (wacban) [reply core/primitives/src/errors.rs:56]: Either way let me remove the From and add more context where it's used currently.

Comment (nagisa) [reply core/primitives/src/errors.rs:56]: well `&'static str` lets you skip all those pesky `into`s in every place where the context is added, that was my primary motivation for suggesting it. And `String` is strictly more general, so if there's a future need for it, it'll be quite straigtforward to make the change, if it ever turns out to be necessary.

But I don't care strongly either way. Removing all those `into`s is also work, so if you don't want to do it, that's fine with me.

Comment (wacban) [reply core/primitives/src/errors.rs:56]: Aah there is so many.. I'll do it next time when I'm debugging those things ;)

**Review by nagisa — APPROVED** [review]

**Pull request merged**